File: v1_21/Reforged/src/main/java/me/fullidle/ficore/ficore/v1_21/PokemonConfigManager.java
package me.fullidle.ficore.ficore.v1_21;

import com.pixelmonmod.pixelmon.api.config.PixelmonConfigProxy;
import me.fullidle.ficore.ficore.common.api.pokemon.IPokemonConfigManager;

import java.util.UUID;

public class PokemonConfigManager implements IPokemonConfigManager {
    public static PokemonConfigManager INSTANCE = new PokemonConfigManager();
    public static final UUID uuid = UUID.randomUUID();

    @Override
    public int getPokeMaxLevel() {
        return PixelmonConfigProxy.getGeneral().getMaxLevel();
    }

    @Override
    public int getComputerBoxes() {
        return PixelmonConfigProxy.getStorage().getComputerBoxes(uuid);
    }
}

File: v1_21/Cob/src/main/kotlin/me/fullidle/ficore/ficore/vCob_21/PokemonConfigManager.kt
package me.fullidle.ficore.ficore.vCob_21

import com.cobblemon.mod.common.Cobblemon
import me.fullidle.ficore.ficore.common.api.pokemon.IPokemonConfigManager

object PokemonConfigManager : IPokemonConfigManager {
    override fun getPokeMaxLevel(): Int {
        return Cobblemon.config.maxPokemonLevel
    }

    override fun getComputerBoxes(): Int {
        return Cobblemon.config.defaultBoxCount
    }
}
